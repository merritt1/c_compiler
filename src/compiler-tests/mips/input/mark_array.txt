  void syscall_print_int(int);
  void syscall_print_string(char*);
  int main(void) {
      int i, a[10];

      for (i = 0; i < 10; i = i + 1) {
          a[i] = i * i;
      }

      for (i = 0; i < 10; i = i + 1) {
          syscall_print_int(a[i]);
          syscall_print_string("\n");
      }

      return 0;
  }

