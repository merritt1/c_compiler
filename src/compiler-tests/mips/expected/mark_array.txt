	.data
_StringLabel_0: .asciiz "\n"

	.text
	.globl main
main:
	addiu $sp, $sp, -144
	sw $fp, 48($sp)
	or $fp, $sp, $0
	sw $ra, 52($fp)
	sw $a0, 0($fp)
	sw $a1, 4($fp)
	sw $a2, 8($fp)
	sw $a3, 12($fp)
	sw $s0, 16($fp)
	sw $s1, 20($fp)
	sw $s2, 24($fp)
	sw $s3, 28($fp)
	sw $s4, 32($fp)
	sw $s5, 36($fp)
	sw $s6, 40($fp)
	sw $s7, 44($fp)
	sw $a0, 96($fp)
	la $s0, 96($fp)
	li $s1, 0
	sw $s1, 0($s0)
_GeneratedLabel_1:
	la $s2, 96($fp)
	li $s3, 10
	lw $s4, 0($s2)
	slt $s5, $s4, $s3
	beqz $s5, _GeneratedLabel_2
	la $s6, 100($fp)
	la $s7, 96($fp)
	lw $t0, 0($s7)
	li $t1, 4
	mul $t2, $t0, $t1
	addu $t3, $s6, $t2
	la $t4, 96($fp)
	la $t5, 96($fp)
	lw $t6, 0($t4)
	lw $t7, 0($t5)
	mul $t8, $t6, $t7
	sw $t8, 0($t3)
	la $t9, 96($fp)
	la $s0, 96($fp)
	li $s1, 1
	lw $s2, 0($s0)
	addu $s3, $s2, $s1
	sw $s3, 0($t9)
	b _GeneratedLabel_1
_GeneratedLabel_2:
	la $s4, 96($fp)
	li $s5, 0
	sw $s5, 0($s4)
_GeneratedLabel_3:
	la $s6, 96($fp)
	li $s7, 10
	lw $t0, 0($s6)
	slt $t1, $t0, $s7
	beqz $t1, _GeneratedLabel_4
	la $t2, 100($fp)
	la $t3, 96($fp)
	lw $t4, 0($t3)
	li $t5, 4
	mul $t6, $t4, $t5
	addu $t7, $t2, $t6
	lw $t8, 0($t7)
	or $a0, $t8, $0
	li $v0,1
	syscall
	la $t9, _StringLabel_0
	or $a0, $t9, $0
	li $v0,4
	syscall
	la $s0, 96($fp)
	la $s1, 96($fp)
	li $s2, 1
	lw $s3, 0($s1)
	addu $s4, $s3, $s2
	sw $s4, 0($s0)
	b _GeneratedLabel_3
_GeneratedLabel_4:
	li $s5, 0
	or $v0, $s5, $0
	b _GeneratedLabel_5
_GeneratedLabel_5:
	li $v0,10
	syscall
